url20b0f880a5_jgraph_legacy_jgraph5_tgz-pJ8-org_jgraph_example_HelloWorldJ8/src/org/jgraph/plaf/basic/BasicGraphUI.java

```diff
@@ -0,0 +1,3220 @@
+				for (double y = ys; y <= ye; y += sgs) {
+					int ix = (int) Math.round(x);
+					int iy = (int) Math.round(y);
+
+					g.drawLine(ix - cs, iy, ix + cs, iy);
+					g.drawLine(ix, iy - cs, ix, iy + cs);
+
+				}
+			}
+
+		}
+			break;
+
+		case JGraph.LINE_GRID_MODE: {
+
+			xe += (int) Math.ceil(sgs);
+			ye += (int) Math.ceil(sgs);
+
+			for (double x = xs; x <= xe; x += sgs) {
+
+				int ix = (int) Math.round(x);
+
+				g.drawLine(ix, ys, ix, ye);
+
+			}
+
+			for (double y = ys; y <= ye; y += sgs) {
+
+				int iy = (int) Math.round(y);
+
+				g.drawLine(xs, iy, xe, iy);
+
+			}
+
+		}
+			break;
+
+		case JGraph.DOT_GRID_MODE:
+		default:
+			for (double x = xs; x <= xe; x += sgs) {
+				for (double y = ys; y <= ye; y += sgs) {
+
+					int ix = (int) Math.round(x);
+					int iy = (int) Math.round(y);
+
+					g.drawLine(ix, iy, ix, iy);
+
+				}
+			}
+			break;
+
+		}
+
+	}
+
+	//
+	// Foreground
+	//
+
+	/**
+	 * Paint the foreground of this graph. Calls paintPorts.
+	 */
+	protected void paintForeground(Graphics g) {
+		if (graph.isPortsVisible() && graph.isPortsOnTop())
+			paintPorts(g, graphLayoutCache.getPorts());
+	}
+
+	/**
+	 * Paint <code>ports</code>.
+	 */
+	public void paintPorts(Graphics g, CellView[] ports) {
+		if (ports != null) {
+			Rectangle r = g.getClipBounds();
+			for (int i = 0; i < ports.length; i++) {
+				if (ports[i] != null) {
+					Rectangle2D tmp = ports[i].getBounds();
+					Rectangle2D bounds = new Rectangle2D.Double(tmp.getX(), tmp
+							.getY(), tmp.getWidth(), tmp.getHeight());
+					Point2D center = new Point2D.Double(bounds.getCenterX(),
+							bounds.getCenterY());
+					if (!graph.isPortsScaled())
+						center = graph.toScreen(center);
+					bounds.setFrame(center.getX() - bounds.getWidth() / 2,
+							center.getY() - bounds.getHeight() / 2, bounds
+									.getWidth(), bounds.getHeight());
+					if (r == null || bounds.intersects(r))
+						paintCell(g, ports[i], bounds, false);
+				}
+			}
+		}
+	}
+
+	//
+	// Various local methods
+	//
+
+	/**
+	 * Update the handle using createHandle.
+	 */
+	public void updateHandle() {
+		if (graphLayoutCache != null) {
+			Object[] cells = graphLayoutCache.getVisibleCells(graph
+					.getSelectionCells());
+			if (cells != null && cells.length > 0)
+				handle = createHandle(createContext(graph, cells));
+			else
+				handle = null;
+		}
+	}
+
+	protected GraphContext createContext(JGraph graph, Object[] cells) {
+		return new GraphContext(graph, cells);
+	}
+
+	/**
+	 * Constructs the "root handle" for <code>context</code>.
+	 * 
+	 * @param context
+	 *            reference to the context of the current selection.
+	 */
+	public CellHandle createHandle(GraphContext context) {
+		if (context != null && !context.isEmpty() && graph.isEnabled()) {
+			try {
+				return new RootHandle(context);
+			} catch (HeadlessException e) {
+				// Assume because of running on a server
+			} catch (RuntimeException e) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: An exception is caught and re-thrown within the catch clause without any other operations being performed on the exception, making the catch clause unnecessary.

**Fix**: Perform additional operations on the exception that has been caught. You can wrap a standard exception into a custom exception to add additional information or special handling. When you do that, make sure to set the original exception as the cause. Otherwise, you will lose the stack trace and message of the original exception, which will make it difficult to analyze the event that caused the exception.  
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)